tools/rv_histogram.c

```diff
@@ -0,0 +1,384 @@
+        X(mulhu)                   \
+        X(div)                     \
+        X(divu)                    \
+        X(rem)                     \
+        X(remu)                    \
+    )                              \
+    IIF(RV32_HAS(EXT_A))(          \
+        X(lrw)                     \
+        X(scw)                     \
+        X(amoswapw)                \
+        X(amoaddw)                 \
+        X(amoxorw)                 \
+        X(amoandw)                 \
+        X(amoorw)                  \
+        X(amominw)                 \
+        X(amomaxw)                 \
+        X(amominux)                \
+        X(amomaxux)                \
+    )                              \
+    IIF(RV32_HAS(EXT_F))(          \
+        X(flw)                     \
+        X(fsw)                     \
+        X(fmadds)                  \
+        X(fmsubs)                  \
+        X(fnmsubs)                 \
+        X(fnmadds)                 \
+        X(fadds)                   \
+        X(fsubs)                   \
+        X(fmuls)                   \
+        X(fdivs)                   \
+        X(fsqrts)                  \
+        X(fsgnjs)                  \
+        X(fsgnjns)                 \
+        X(fsgnjxs)                 \
+        X(fmins)                   \
+        X(fmaxs)                   \
+        X(fcvtws)                  \
+        X(fcvtwus)                 \
+        X(fmvxw)                   \
+        X(feqs)                    \
+        X(flts)                    \
+        X(fles)                    \
+        X(fclasss)                 \
+        X(fcvtsw)                  \
+        X(fcvtswu)                 \
+        X(fmvmx)                   \
+    )                              \
+    IIF(RV32_HAS(EXT_C))(          \
+        X(caddi4spn)               \
+        X(clw)                     \
+        X(csw)                     \
+        X(cnop)                    \
+        X(caddi)                   \
+        X(cjal)                    \
+        X(cli)                     \
+        X(caddi16sp)               \
+        X(clui)                    \
+        X(csrli)                   \
+        X(csrai)                   \
+        X(candi)                   \
+        X(csub)                    \
+        X(cxor)                    \
+        X(cor)                     \
+        X(cand)                    \
+        X(cj)                      \
+        X(cbeqz)                   \
+        X(cbnez)                   \
+        X(cslli)                   \
+        X(clwsp)                   \
+        X(cjr)                     \
+        X(cmv)                     \
+        X(cebreak)                 \
+        X(cjalr)                   \
+        X(cadd)                    \
+        X(cswsp)                   \
+    )                              \
+    X(unknown)
+/* clang-format on */
+
+static rv_insn_hist_t rv_insn_stats[] = {
```

Comment:
Remove the proposed `RISCV_INSN_STATS` macro and rewrite as following:
```c
static rv_insn_hist_t rv_insn_stats[] = {
#define _(inst, can_branch) {#inst, 0},
        RISCV_INSN_LIST
#undef _                                                                                                                                                      
};
```

Be mindful of the notation: "insn" is employed for mnemonic identifiers, while "inst" is used for parameters.

Reply:
Great technique for reusing `RISCV_INSN_LIST` !